File: app/src/main/java/com/weatherkremenchug/weatherkremenchugtest/retrofit/CallToServer.java
package com.weatherkremenchug.weatherkremenchugtest.retrofit;

import com.weatherkremenchug.weatherkremenchugtest.data.WeatherForecast;
import com.weatherkremenchug.weatherkremenchugtest.data.WeatherOnToday;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface CallToServer {

    @GET("forecast")
    Call<WeatherForecast> getForecast(@Query("lat") Double lat,
                                      @Query("lon") Double lon,
                                      @Query("units") String units,
                                      @Query("appid") String appid);

    @GET("weather")
    Call<WeatherOnToday> getWeather(@Query("lat") Double lat,
                                    @Query("lon") Double lon,
                                    @Query("units") String units,
                                    @Query("appid") String appid);

}

File: app/src/main/java/com/weatherkremenchug/weathermain/retrofit/CallToServer.kt
package com.weatherkremenchug.weathermain.retrofit

import com.weatherkremenchug.weathermain.data.WeatherForecast
import com.weatherkremenchug.weathermain.data.WeatherOnToday

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface CallToServer {

    @GET("forecast")
    fun getForecast(@Query("lat") lat: Double?,
                    @Query("lon") lon: Double?,
                    @Query("units") units: String,
                    @Query("appid") appid: String): Call<WeatherForecast>

    @GET("weather")
    fun getWeather(@Query("lat") lat: Double?,
                   @Query("lon") lon: Double?,
                   @Query("units") units: String,
                   @Query("appid") appid: String): Call<WeatherOnToday>

}